payments-core/src/main/java/com/stripe/android/StripeIntentResult.kt

```diff
@@ -30,27 +30,51 @@ abstract class StripeIntentResult<out T : StripeIntent> internal constructor(
             return outcome
         }
 
-        when (stripeIntentStatus) {
-            StripeIntent.Status.RequiresAction, StripeIntent.Status.Canceled -> {
-                return Outcome.CANCELED
+        return when (stripeIntentStatus) {
+            StripeIntent.Status.RequiresAction -> {
+                if (isNextActionSuccessState(intent)) {
+                    Outcome.SUCCEEDED
+                } else {
+                    Outcome.CANCELED
```

Comment:
are we sure we want `DisplayOxxoDetails` and `MicroDeposits` result in a `CANCELLED` outcome here? If so, I'd add more documentation around `isNextActionSuccessState` to explain it.